tests/serving-functions.test.js

```diff
@@ -474,4 +477,42 @@ test('Serves functions that dynamically load files included in the `functions.in
     })
   })
 })
+
+test('Uses sourcemaps to show correct paths and locations in stack trace', async (t) => {
+  await withSiteBuilder('function-with-sourcemaps', async (builder) => {
+    await builder
+      .withFunction({
+        path: 'hello.js',
+        handler: async () => {
+          throw new Error('Something went wrong')
+        },
+      })
+      .withNetlifyToml({
+        config: {
+          build: { publish: 'public' },
+          functions: { directory: 'functions', node_bundler: 'esbuild' },
+        },
+      })
+      .buildAsync()
+
+    await withDevServer({ cwd: builder.directory }, async ({ port }) => {
+      const sourcemap = await readFileAsync(
+        join(builder.directory, '.netlify', 'functions-serve', 'hello', 'src', 'hello.js.map'),
+        'utf8',
+      )
+
+      console.log('-> Sourcemap', sourcemap)
```

Comment:
I've used these `console.log` statements to debug a test failure in the CI. I'll remove them before merging.